src/ResultsHelpers.ts

```diff
@@ -0,0 +1,839 @@
+        } else {
+          statementResult.pretty = 'UNHIT';
+        }
+      }
+    } else if (doesResultPass(rawStatementResult)) {
+      statementResult.final = FinalResult.TRUE;
+      if (doPretty) {
+        statementResult.pretty = prettyResult(rawStatementResult);
+      }
+    } else {
+      statementResult.final = FinalResult.FALSE;
+      if (rawStatementResult instanceof Array && rawStatementResult.length === 0) {
+        // Special case, handle empty array.
+        if (doPretty) {
+          statementResult.pretty = 'FALSE ([])';
+        }
+      } else if (MeasureHelpers.isStatementFunction(elmLibrary, statementResult.statementName)) {
+        if (doPretty) {
+          statementResult.pretty = 'FUNCTION';
+        }
+      } else if (doPretty) {
+        statementResult.pretty = `FALSE (${rawStatementResult})`;
+      }
+    }
+
+    if (includeClauseResults) {
+      // create clause results for all localIds in this statement
+      const localIds = MeasureHelpers.findAllLocalIdsInStatementByName(elmLibrary, statementResult.statementName);
+      for (const localId in localIds) {
+        const clause = localIds[localId];
+        const rawClauseResult =
+          rawClauseResults[statementResult.libraryName] != null
+            ? rawClauseResults[statementResult.libraryName][
+                clause.sourceLocalId != null ? clause.sourceLocalId : localId
+              ]
+            : undefined;
+
+        const clauseResult: ClauseResult = {
+          // if this clause is an alias or a usage of alias it will get the raw result from the sourceLocalId.
+          raw: rawClauseResult,
+          statementName: statementResult.statementName,
+          libraryName: statementResult.libraryName,
+          localId,
+          final: FinalResult.NA
+        };
+
+        clauseResult.final = setFinalResults({
+          rawClauseResults,
+          statementRelevance: statementResult.relevance,
+          libraryName: statementResult.libraryName,
+          clause,
+          rawResult: clauseResult.raw
+        });
+        clauseResults.push(clauseResult);
+      }
+    }
+  });
+
+  return clauseResults;
+}
+
+/**
+ * Generates a pretty human readable representation of a result.
+ *
+ * @param {(Array|object|boolean|???)} result - The result from the calculation engine.
+ * @param {number|undefined} indentLevel - For nested objects, the indentLevel indicates how far to indent.
+ *                                Note that 1 is the base because Array(1).join ' ' returns ''.
+ * @param {number|undefined} keyIndent - Indent count used for key indentation.
+ * @returns {String} a pretty version of the given result
+ */
+export function prettyResult(result: any | null, indentLevel?: number, keyIndent?: number): string {
+  // TODO: Sort out a better way to have a friendly system display for FHIR codes.
+  //   This will need to be replaced with a canonical URL to code system name map.
+  const nameOidHash = {
```

Comment:
According to my editor, this variable isn't used anywhere. Is this here for future-proofing, or should we remove it?

Reply:
The TODO there explains the deal. We need to replace it with something more FHIR friendly. I think this would be good further work if we want to make use of the the pretty results from statements. Is it okay to leave this for now with the TODO?

Reply:
Yeah, that makes sense. As long as there's a reason for it to be here, it's g2g.